packages/swing-store/src/swingStore.js

```diff
@@ -525,6 +541,37 @@ function makeSwingStore(dirPath, forceReset, options) {
     stopTrace();
   }
 
+  /**
+   * Return a Buffer with the entire DB state, useful for cloning a
+   * small swingstore in unit tests.
+   *
+   * @returns {Buffer}
+   */
+  function serialize() {
+    // An on-disk DB with WAL mode enabled seems to produce a
+    // serialized Buffer that can be unserialized, but the resulting
+    // 'db' object fails all operations with SQLITE_CANTOPEN. So
+    // pre-emptively throw.
+    if (filePath !== ':memory:') {
+      throw Error('on-disk DBs with WAL mode enabled do not serialize well');
+    }
```

Comment:
Should this be an assert?

Reply:
Hm, I'm unsure.. it's not like the user did something wrong, it's more like "I wish I could accomodate that, but I can't, sorry". `assert.meekly()`, as it were. I guess I'll leave it.